web/assets/js/components/dashboard.ts

```diff
@@ -1,102 +1,112 @@
-  }
-
-  async _feedRatings() {
-    const ratings = await Promise.all([this._feedUserBlitzRating(), this._feedUserRapidRating()]);
-    const globalRating = ratings.reduce((acc, curr) => {
-      curr.win += acc.win;
-      curr.lose += acc.lose;
-      curr.draws += acc.draws;
-      return curr;
-    });
-    this._feedUserGlobalRating(globalRating);
-  }
-
-  _feedLeaderbord() {
-    const leaders = [Actions.getBlitzLeaders(), Actions.getRapidLeaders()];
-
-    //TODO: check tailwind classes
-    const leaderMapped = leaders.map((leadmap) => {
-      return leadmap
-        .map((lead: any) => {
-          return `<li class="dashboard-avatar flex">
+    userToken: string | undefined | null;
+
+    constructor(opts: any) {
+        super(opts);
+    }
+
+    init() {
+        // automatically called at start
+        console.log("Dashboard component init");
+
+        //TODO: login/logout - redir
+        //TODO: feed data contents - personal
+        //TODO: feed data contents - leaderboard (dedicated component ?)
+
+        Actions.getInstance().then((actions: Actions) => {
+            const userToken = actions.getFaucetToken()
+            if (!userToken) {
+                // TODO handle
+                throw new Error("user token not present")
+            }
+
+            this.userToken = userToken
+            this._feedUser(this.userToken);
+            this._feedRatings();
+            this._feedLeaderbord();
+        }).catch(() => {
+            // TODO handle error
+        })
+    }
+
+    _feedUserSocial() {
+    }
+
+    _feedUser(token: string) {
+        //avatar
+        //TODO: avatar creation
+
+        //token
+        const DOM = document.getElementById("js-dashtoken");
+        if (DOM && token) DOM.innerHTML = truncateString(token, 4, 4);
+    }
+
+    _createDomFunc(category: Categoy, rating: any) {
+        const kinds = ["wins", "draws", "loses", "game"];
+        const games = document.getElementById(`js-dashboard${category}Games`);
+
+        kinds.forEach((kind) => {
+            const el = document.getElementById(`js-dashboard${category}${kind.charAt(0).toUpperCase() + kind.slice(1)}`);
+            if (el) el.innerHTML = rating[kind].toString();
+        });
+        if (games) games.innerHTML = (rating.loses + rating.draws + rating.wins).toString();
+
+        return rating;
+    }
+
+    _feedUserGlobalRating(globalRating: any) {
+        return this._createDomFunc("Global", globalRating);
+    }
+
+    async _feedRatings() {
+        const ratings = await Promise.all([this._feedUserBlitzRating(), this._feedUserRapidRating()]);
+        const globalRating = ratings.reduce((acc, curr) => {
+            curr.win += acc.win;
+            curr.lose += acc.lose;
+            curr.draws += acc.draws;
+            return curr;
+        });
+        this._feedUserGlobalRating(globalRating);
+    }
+
+    _feedLeaderbord() {
+        // TODO @Alexis
+        // There is no Blitz / Rapid leaderboard,
```

Comment:
I'm confused, if there is no way to get leaderboard by category, how can we decide winner by game category in the event contest?

Reply:
@thehowl